src/Time/Actions/RecordTimeStepperData.hpp

```diff
@@ -47,7 +47,7 @@ namespace Actions {
 /// - Modifies:
 ///   - dt_variables_tag,
 ///   - Tags::HistoryEvolvedVariables<variables_tag>
-template <typename VariablesTag = NoSuchType>
+template <typename move_dt = std::true_type, typename VariablesTag = NoSuchType>
```

Comment:
first template parameter should be `bool MoveTimeDerivative`

Reply:
@wthrowe can you look at the changes to this action in this commit (i.e. don't review the whole PR)

Reply:
I've been preferring template parameters to be types when they can be, because it makes the template classes generally more amenable to manipulation via brigand (e.g. in binds and within lambdas). Why prefer a `bool`?

Reply:
Because it only needs to be a bool, and it's a lot easier to type... I don't think we should change our code's clarity for the "Oh, we may need to use this in a template-template parameter context to extract template parameters". You also won't need to wrap this in a `bind` because it would already be a lazy-metafunction.

Reply:
The interface to the History class doesn't really make sense anymore.  The internals have evolved over time and the only thing the argument being taken by rvalue reference avoids is copying the data array (not the object, just doing a `memcpy` or similar).  This is probably not worth the complexity.

Change this to always copy, and I'll submit a PR to change the interface to take by const reference.  Whichever PR gets merged first will cause an easy merge conflict on this file for the other.